packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,295 @@
+            };
+
+            await expect(sometimes)
+                .retry()
+                .to.deep.equal({ c: { b: { a: 1 } } });
+        });
+
+        it('.nested', async () => {
+            let attempt = 0;
+            const sometimes = () => {
+                attempt += 1;
+                return attempt !== 5 ? null : { a: { b: ['x', 'y'] } };
+            };
+
+            await expect(sometimes).retry().to.have.nested.property('a.b[1]');
+        });
+    });
+
+    describe('should pass the assertions that ends with properties:', () => {
+        it('.null', async () => {
+            let attempt = 0;
+            const sometimesNullFunction = () => {
+                attempt += 1;
+                return attempt !== 5 ? null : 'not-null';
+            };
+
+            await expect(sometimesNullFunction).retry({ timeout: 2000 }).to.be.not.null;
+
+            expect(attempt).to.equal(5);
+        });
+
+        it('.undefined', async () => {
+            let attempt = 0;
+            const sometimesUndefinedFunction = () => {
+                attempt += 1;
+                return attempt !== 5 ? 'not-undefined' : undefined;
+            };
+
+            await expect(sometimesUndefinedFunction).retry({ timeout: 2000 }).to.be.undefined;
+
+            expect(attempt).to.equal(5);
+        });
+
+        it('.empty', async () => {
+            let attempt = 0;
+            const sometimes = () => {
+                attempt += 1;
+                return attempt !== 5 ? [1, 2, 3] : [];
+            };
+
+            await expect(sometimes).retry({ retries: 10 }).to.be.empty;
+
+            expect(attempt).to.equal(5);
+        });
+
+        it('.sealed', async () => {
+            const sealedObject = Object.seal({ prop1: 'value1', prop2: 'value2' });
+
+            await expect(() => sealedObject).retry().to.be.sealed;
+        });
+
+        it('.not.sealed', async () => {
+            const notSealedObject = { prop1: 'value1', prop2: 'value2' };
+
+            try {
+                await expect(() => notSealedObject).retry({ retries: 3 }).to.be.sealed;
+            } catch (error: unknown) {
+                expect(error).to.be.an.instanceOf(Error);
+                expect((error as Error).message).to.include('3 retries');
+            }
+        });
+    });
+
+    describe('should work with various assertion methods:', () => {
+        it('.satisfy()', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
+                attempts++;
+                return { attempts };
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 5, delay: 10 })
+                .to.satisfy((obj: { attempts: number }) => obj.attempts > 3);
+
+            expect(attempts).to.equal(4);
+        });
+
+        it('.have.property(), and.be.above()', async () => {
+            let attempts = 0;
+
+            const funcToRetry = () => {
```

Comment:
let's write some helper to make function return one value before certain amount of calls and another after.

Reply:
@yurii-ve I think it would not be really useful cause I almost always need to check attempts

of course I could return attempts from function but that means to change all of tests 😅

Reply:
I don't get why do you alway need to check attempts? The point is to check if test passes when it have to pass and fails when it have to fail. In my opinion you need only one test to check if correct amount of attempts had been made.

Reply:
@yurii-ve but it already helped me numerous of times in different tests to identify that plugin failing to work as expected due to pass/fail on the wrong attempt, why should we make us more vulnerable to possible bugs?

Reply:
@yurii-ve added helper function